src/config.py

```diff
@@ -19,21 +18,39 @@
 # build directory. Debug symbols and __FILE__ use absolute paths.
 DIR_BUILDROOT = '/usr/obj/cloudabi-ports'
 
-# Location where resource files are stored.
-DIR_RESOURCES = os.path.join(os.getcwd(), 'misc')
+
+RESOURCES = dict((name, get_data('misc', name))
+                 for name in ['config.guess', 'config.sub'])
 
 # Location at which distfiles can be fetched in case the master sites
 # are down.
 FALLBACK_MIRRORS = {'https://nuxi.nl/distfiles/third_party/'}
 
-# Host C and C++ compiler, used to compile the build tools. We'd better
-# use Clang if available. Compared to GCC, it has the advantage that it
-# does not depend on the 'as' and 'ld' utilities being part of $PATH.
-HOST_CC = ('/usr/bin/clang-3.7' if platform.system() == 'Linux' else
-           '/usr/bin/cc')
-HOST_CXX = ('/usr/bin/clang++-3.7' if platform.system() == 'Linux' else
-            '/usr/bin/c++')
-
-# Name of the Perl executable.
-PERL = ('/usr/local/bin/perl' if platform.system() == 'FreeBSD' else
-        '/usr/bin/perl')
+
+class Platform(object):
```

Comment:
Perhaps; I haven't really thought that one through. I was just in ocap-refactor mode, not thinking much. I didn't start to understand what the code really does until much later.

Perhaps I should explain the technique a bit more:
 1. Open a python file in emacs with flycheck and flake8
 2. Review module imports; strike any that are not [powerless][] such as `import os`, `import sys`, platform (these will be imported in `_script` later)
 3. C-!-n (flycheck-next-error) to visit each `undefined name` error and salt to taste
     - Add arguments to pass capabilities explicitly
       - add doctests (or static type declarations using [py2scala][] or mypy) to gain confidence that the refactoring is correct (_I haven't started on this part._)
 4. As noted in [Capability idioms for python][1], if there is any code not in a function, put it in a `main` function and add call it from the top-level script environment: `if __name__ == '__main__'`
    - encapsulate powerful imports in `_script()`
      - Note that the `_initial_caps()` idiom in that 2013 blog item isn't quite right; it leaves the powerful capabilities un-encapsulated

[1]: http://www.madmode.com/2013/python-capability-idioms-part-1.html
[powerless]: http://waterken.sourceforge.net/javadoc/org/joe_e/Powerless.html
[py2scala]: https://bitbucket.org/DanC/py2scala